pandas/tools/plotting.py

```diff
@@ -2872,12 +2875,14 @@ def grouped_hist(data, column=None, by=None, ax=None, bins=50, figsize=None,
     -------
     axes: collection of Matplotlib Axes
     """
-    def plot_group(group, ax):
-        ax.hist(group.dropna().values, bins=bins, **kwargs)
+    def plot_group(group, ax, weights=None):
+        if weights is not None:
+            weights=weights.dropna().values
```

Comment:
ah, issues when na is different on weights and group arrays, did not think of this, will think a little bit more